src/SymfonyCache/EventDispatchingHttpCache.php

```diff
@@ -89,28 +93,25 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
      */
     protected function store(Request $request, Response $response)
     {
-        if ($this->getEventDispatcher()->hasListeners(Events::PRE_STORE)) {
-            $event = new CacheEvent($this, $request, $response);
-            $this->getEventDispatcher()->dispatch(Events::PRE_STORE, $event);
-            $response = $event->getResponse();
-        }
+        $response = $this->dispatch(Events::PRE_STORE, $request, $response);
 
         parent::store($request, $response);
     }
 
     /**
-     * Dispatch the POST_HANDLE event if needed.
+     * Dispatch an event if needed.
      *
-     * @param Request  $request
-     * @param Response $response
+     * @param string        $name     Name of the event to trigger. One of the constants in FOS\HttpCache\SymfonyCache\Events
+     * @param Request       $request
+     * @param Response|null $response If already available
      *
-     * @return Response The response to return which might be altered by a POST_HANDLE listener.
+     * @return Response The response to return, which might be provided/altered by a listener.
      */
-    private function dispatchPostHandle(Request $request, Response $response)
+    protected function dispatch($name, Request $request, Response $response = null)
```

Comment:
made this protected so that kernels using the trait can also use it, if they need to expand on a method and still need the pre and post event.